scripts/export_event_types.py

```diff
@@ -45,71 +77,119 @@
+                    n_types=n_types,
+                    return_partition=True,
+                )
+            else:
+                d_types, event_type_partition = event_types.partition_event_types(
+                    dtf_test,
+                    labels=labels,
+                    log_e_bins=event_type_log_e_bins,
+                    n_types=n_types,
+                    return_partition=True,
+                )
         else:
-            # Calculate event types for proton and electron events,
-            # using the same event type thresholds as in the gamma-like gammas:
-            dtf_e_formatted = {"default": dtf_e_test}
-            dtf_test = event_types.add_predict_column(dtf_e_formatted, trained_model)
-            d_types = event_types.partition_event_types(
-                dtf_test,
-                labels=labels,
-                log_e_bins=event_type_log_e_bins,
-                n_types=n_types,
-                event_type_bins=event_type_partition,
-            )
-        # Start creating the event_type column within the original dataframe:
-        dtf["event_type"] = -99
-        # from IPython import embed
-        # embed()
-        for energy_key in dtf_e_test.keys():
-            if "gamma_cone" in dl2_file:
+            # Calculate event types for proton and electron events, using the same event type
+            # thresholds as in the gammas:
+            if offset_partition:
+                d_types = event_types.partition_event_types_offsetwise(
+                    dtf_test,
+                    labels=labels,
+                    log_e_bins=event_type_log_e_bins,
+                    offset_bins=event_type_offset_bins,
+                    n_types=n_types,
+                    event_type_bins=event_type_partition,
+                )
+            else:
+                d_types = event_types.partition_event_types(
+                    dtf_test,
+                    labels=labels,
+                    log_e_bins=event_type_log_e_bins,
+                    n_types=n_types,
+                    event_type_bins=event_type_partition,
+                )
+        # Start creating the event_type column within the original dataframe,
+        # setting the default value: -3.
+        # There shouldn't be many -3s at the end. If that's the case, it must be understood.
+        dtf["event_type"] = -3
+
+        if particle is gamma:
+            # Assign type -1 to the events used for the training.
+            for energy_key in dtf_e_train.keys():
                 dtf.loc[dtf_e_train[energy_key].index.values, "event_type"] = -1
-        dtf.loc[dtf_test["default"].index.values, "event_type"] = dtf_test["default"]["event_type"]
+
+        dtf.loc[dtf_test[suffix].index.values, "event_type"] = dtf_test[suffix]["event_type"]
 
         print("A total of {} events will be written.".format(len(dtf["event_type"])))
-        for event_type in [-99, 1, 2, 3]:
+        dtf_7 = dtf[dtf["cut_class"] != 7]
```

Comment:
I assume this is done only for reporting purposes and not actually used, right?

Reply:
Yes, that is just to check if the statistics are even when you only count the gamma-like events, but that doesn't affect anything else, all the training and event-type partitioning is done with all the events.